### ISSUE TITLE
Question - how to limit access the published notes to only signed-in users?

### ISSUE THREAD

**Issue by zulhfreelancer** [hackmdio/codimd#1402]
I have published a note to my self-hosted CodiMD instance and the shareable link looks like https://example.com/s/4OtuyRvuc

When I viewed the link in another browser (I'm not logged in as CodiMD user), I can see the note. I was expecting an error message saying I don't have view permission for that note.

I tried changing the note permissions (on owner side) but none of them worked. I'm still able to see the published notes on non-logged in browser.

How to only whitelist signed-in users to see the published notes? Did I missed something?

**Comment by zulhfreelancer** [comment]
@jackycute appreciate if you can spare a minute to answer this :)

**Comment by jackycute** [comment]
Hi @zulhfreelancer,

You need to change the note permission to either `Limited`, `Protected`.
For the permission details, please see: https://codimd-app.herokuapp.com/features#Permissions

**Comment by zulhfreelancer** [comment]
@jackycute I already tried it. The published note is still visible to not-logged in users.

**Comment by jackycute** [comment]
Can you reproduce that on https://codimd-app.herokuapp.com?
I tried this: https://codimd-app.herokuapp.com/hOoY0WwIR62WpcIyoO44-g
And I use incognito mode to visit that, got proper error message.

<img width="1028" alt="螢幕快照 2020-02-07 上午10 05 16" src="https://user-images.githubusercontent.com/2639151/73994741-93fae380-4991-11ea-821c-012628d5e3c8.png">

**Comment by zulhfreelancer** [comment]
Here is my tests @jackycute —

Test 1

![Screen Shot 2020-02-07 at 10 13 28 AM](https://user-images.githubusercontent.com/3416976/73995090-9ad62600-4992-11ea-94e9-72aca79432d8.png)

![Screen Shot 2020-02-07 at 10 13 50 AM](https://user-images.githubusercontent.com/3416976/73995098-a1fd3400-4992-11ea-9dab-c397160f310c.png)

---

Test 2

![Screen Shot 2020-02-07 at 10 13 05 AM](https://user-images.githubusercontent.com/3416976/73995129-b5100400-4992-11ea-8b0f-142b8ab83aff.png)

![Screen Shot 2020-02-07 at 10 12 53 AM](https://user-images.githubusercontent.com/3416976/73995151-bfca9900-4992-11ea-9c23-a518e9839c98.png)

**Comment by jackycute** [comment]
@zulhfreelancer I see that now.
The published page won't be restricted, sounds like a bug. We will fix soon.

**Comment by zulhfreelancer** [comment]
@jackycute thank you for confirming. Appreciate it!

**Comment by jackycute** [comment]
Hey @zulhfreelancer could you use version 1.4.1? this bug should only in the develop branch now.

**Comment by zulhfreelancer** [comment]
@jackycute it works now. Thank you!

**Comment by zulhfreelancer** [comment]
Hey @jackycute. I'm reopening this issue because I think the bug is still there.

---

Test 3 - https://codimd-app.herokuapp.com/s/YOasVE8A0

The permission is _Editable - Signed-in people can edit_ but non-logged in user still able to see the published note (link above).

I was expecting the non-logged in user to see _You're not allowed to see..._ error.

![Screen Shot 2020-02-07 at 6 25 44 PM](https://user-images.githubusercontent.com/3416976/74022503-a2b8b900-49d8-11ea-8678-82b782fb9471.png)

![Screen Shot 2020-02-07 at 6 26 49 PM](https://user-images.githubusercontent.com/3416976/74022514-a8160380-49d8-11ea-891f-8f4226febebd.png)

![Screen Shot 2020-02-07 at 6 32 20 PM](https://user-images.githubusercontent.com/3416976/74022517-a9dfc700-49d8-11ea-8616-12f39f34e4fa.png)

---

Test 4 - https://codimd-app.herokuapp.com/s/T3UWaMRH1

The permission is _Locked - Only owner can edit_ but non-logged in user still able to see the published note (link above).

I was expecting the non-logged in user to see _You're not allowed to see..._ error.

![Screen Shot 2020-02-07 at 6 27 24 PM](https://user-images.githubusercontent.com/3416976/74022855-5fab1580-49d9-11ea-92c6-f5e010735744.png)

![Screen Shot 2020-02-07 at 6 27 48 PM](https://user-images.githubusercontent.com/3416976/74022859-6174d900-49d9-11ea-9fd6-b074607f974e.png)

![Screen Shot 2020-02-07 at 6 32 57 PM](https://user-images.githubusercontent.com/3416976/74022865-62a60600-49d9-11ea-8a60-666d51452517.png)

---

Test 5 - https://codimd-app.herokuapp.com/s/M1ERpTeKb

The permission is _Private - Only owner can view & edit_ and this seems to work as expected. Non-logged in user sees the _You're not allowed to see..._ error when navigating to the editor mode (https://codimd-app.herokuapp.com/pVJbZg8BReWBXW3tD0PXfQ?both) - as expected.

However, when the non-logged in user navigates to https://codimd-app.herokuapp.com/s/M1ERpTeKb, the server is not responding. Looks like it's having internal server error. 

![Screen Shot 2020-02-07 at 6 30 12 PM](https://user-images.githubusercontent.com/3416976/74023153-e7911f80-49d9-11ea-9031-4a9eaf532a57.png)

---

In short, I think the app shouldn't show the note editor and published view to non-logged in user when the note's permission is set to _Editable_ or _Locked_. For signed-in users, that's fine.

Can you confirm if this is a bug?

**Comment by jackycute** [comment]
Hi @zulhfreelancer,

That is by design that:

- Editable can be viewed by anonymous user
- Locked can be viewed by anonymous user

Here is the table of those permission explained:
<img width="743" alt="螢幕快照 2020-02-07 下午10 24 02" src="https://user-images.githubusercontent.com/2639151/74037120-97758580-49f8-11ea-8cee-11e31dfb0e03.png">

We'll take a look at the error log for http://codimd-app.herokuapp.com/s/M1ERpTeKb

**Comment by zulhfreelancer** [comment]
Noted. Thanks @jackycute.

FYI, note with Limited access also triggers internal server error. Check this one out without logged in - https://codimd-app.herokuapp.com/s/00_fesXDp

**Comment by jackycute** [comment]
This should be fixed in #1416.

Feel free to reopen if you find further issue @zulhfreelancer 
Thanks for reporting again.
